packages/odyssey-react/src/components/Tab/index.tsx

```diff
@@ -106,9 +109,51 @@ const Tabs: FunctionComponent<PropsTabs> & StaticComponents = ({ children, id, s
     }
   }
 
+  const handleTabFocus = (event: KeyboardEvent) => {
+    if (tabElement.current) {
+      const focusableSelector = `
+      a[href],
+      button,
+      textarea,
+      input[type="text"],
+      input[type="radio"],
+      input[type="checkbox"],
+      select
+      `;
+      const tabs = tabElement.current.querySelectorAll(focusableSelector);
+      const tabsArr = Array.prototype.slice.call(tabs);
+      const tabCount = tabs.length;
+      const focusableElFirst = tabs[0] as HTMLElement;
+      const focusableElLast = tabs[tabCount - 1] as HTMLElement;
+      const focusableIndexActive = tabsArr.indexOf(document.activeElement);
+      const focusableIndexLast = tabsArr.indexOf(focusableElLast);
+      const focusableElNext = tabs[focusableIndexActive + 1] as HTMLElement;
+      const focusableElPrev = tabs[focusableIndexActive - 1] as HTMLElement;
```

Comment:
why the type assertion here? Doesn't feel good and isn't type safe. Don't we have confidence which elements tabs will be since we're the ones providing the framework? I think you need to use the generic form of `querySelectorAll` above to prevent this:

```
const tabs = tabElement.current.querySelectorAll<HTMLButtonElement>(focusableSelector);
```

Reply:
Fixed.

Reply:
I just took another look at this - I'm not sure my feedback is correct because your `focusableSelector` **can** target a variety of different nodes. Is the intention here to only get tabs (which are HTMLButton under the hood if i'm not mistaken) or a broader set of elements? If its the latter we need to use type refinement